packages/cli-kit/src/private/node/session/device-authorization.ts

```diff
@@ -47,11 +50,25 @@ export async function requestDeviceAuthorization(scopes: string[]): Promise<Devi
 
   outputInfo('\nTo run this command, log in to Shopify.')
   outputInfo(outputContent`User verification code: ${jsonResult.user_code}`)
-  outputInfo(
-    outputContent`👉 Open this link to start the auth process: ${outputToken.green(
-      jsonResult.verification_uri_complete,
-    )}`,
-  )
+  if (isTTY() && !isCloudEnvironment()) {
+    outputInfo('👉 Press any key to open the login page on your browser')
+    await keypress()
+    await openURL(jsonResult.verification_uri_complete)
+    outputInfo(
+      outputContent`Opened link to start the auth process: ${outputToken.link(
+        'Login link',
+        jsonResult.verification_uri_complete,
+      )}`,
+    )
+  } else {
+    outputInfo(outputContent`User verification code: ${jsonResult.user_code}`)
+    outputInfo(
+      outputContent`👉 Open this link to start the auth process: ${outputToken.link(
```

Comment:
I think in this case the `green` approach is better, as we've already established this terminal might not be TTY aware anyway, and it reads a bit better with out the extra formatting we get with `link` in unsupported environments

Reply:
Is that actually true? In cloud environments, are links generally non-clickable?